features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesView.kt

```diff
@@ -168,11 +186,44 @@ fun MessagesView(
         },
     )
 
+    var reactingToEventId: EventId? by remember { mutableStateOf(null) }
     ActionListView(
         state = state.actionListState,
+        sheetState = actionListViewBottomSheetState,
         isVisible = isMessageActionsBottomSheetVisible,
-        onDismiss = ::onDismissActionListBottomSheet,
-        onActionSelected = ::onActionSelected
+        onDismiss = { coroutineScope.launch { onDismissActionListBottomSheet() } },
+        onActionSelected = ::onActionSelected,
+        onCustomReactionClicked = { event ->
+            reactingToEventId = event.eventId
```

Comment:
The data flow here feels a bit weird. Should this be pulled into the state maybe? :thinking:

Reply:
Are you talking about the `reactingToEventId` being declared here? I thought about moving it to some *State class, but that required creating a dedicated event and it didn't seem to be worth it.

Reply:
Yeah, I found it a bit hard to follow where it was being set/unset/used as they're all in callbacks on different components. I think having it in the state with event(s) as needed would be a little clearer, but I don't feel terribly strongly about it -- if you don't think it's worth it then happy for it to go as is :)